src/Composer/Satis/PackageSelection/PackageSelection.php

```diff
@@ -284,6 +290,19 @@ public function load()
     }
 
     /**
+     * Mark packages as abandoned by Configuration file
+     */
+    private function setAbandoned()
```

Comment:
I'm not going to ask you to change this, but I do have a suggestion for future PR's.

Using methods on the class like this internally in the function makes them a teensy bit uglier to test. It is always better to pass in dependencies of methods. It also makes it more clear from the function signature what it works with at that point then. Cause right now, you have to read through the whole function body to determine what it could possibly be doing/changing.